### PR TITLE
[BUGFIX] Prevent evaluating ExpressionNode unless demanded

### PR DESCRIPTION
**Pull request by NamelessCoder**
Adds a new interface which must be used by expression node
types in order to signal that the expression node must be
evaluated during parsing of the template.

History:

Parse-time evaluation of ExpressionNodes is a remnant from
the time when there was a separate legacy namespace
extraction based on expression nodes. While there currently
are expression-like template processors, there are no known
implementations out there which must have the node evaluated
during parse time.

Choices:

Rather than simply remove the evaluation altogether, the new
interface is added as a quicker way to migrate *if* there are
private repositories out there implementing these nodes.
If the evaluation was removed entirely, the only alternative left
would be migrating to a template pre-processor which is by
nature quite different from the expression nodes. So, as a
convenience, this interface is left in place to be implemented
as substitute for the old interface, requiring no further change.

Close: #300

### CODE REVIEW

**Comment by NamelessCoder** [comment]
Alternatives:

1. Demand refactoring old expression nodes which depend on parse-time evaluation.
2. Add this interface as-is and consider parse-time evaluation a (very infrequently used) existing feature.
3. Add this interface, but deprecate it immediately, and inform that parse-time evaluation will be removed in Fluid 3.0

I lean towards 1 or 3. Parse-time evaluation is the domain of template pre-processors. Any thoughts?

**Comment by liayn** [comment]
Option 3 sounds good

**Comment by NamelessCoder** [comment]
Code updated to reflect option 3.

File: src/Core/Parser/SyntaxTree/Expression/ParseTimeEvaluatedExpressionNodeInterface.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+namespace TYPO3Fluid\Fluid\Core\Parser\SyntaxTree\Expression;
+
+/*
+ * This file belongs to the package "TYPO3 Fluid".
+ * See LICENSE.txt that was shipped with this package.
+ */
+
+use TYPO3Fluid\Fluid\Core\Compiler\TemplateCompiler;
+use TYPO3Fluid\Fluid\Core\Parser\SyntaxTree\NodeInterface;
+use TYPO3Fluid\Fluid\Core\Rendering\RenderingContextInterface;
```
Comment (liayn) [inline src/Core/Parser/SyntaxTree/Expression/ParseTimeEvaluatedExpressionNodeInterface.php]: unused?

**Pull request merged**